src/lib.rs

```diff
@@ -42,44 +43,113 @@ use std::thread;
 use tuikit::prelude::{Event as TermEvent, *};
 
 //------------------------------------------------------------------------------
-pub trait SkimItem: Send + Sync {
-    /// The text to be displayed on the item list, could contain ANSI properties
-    fn display(&self) -> Cow<AnsiString>;
-
-    /// helper function to get pure text presentation(without color) of the item
-    fn get_text(&self) -> Cow<str>;
+pub trait AsAny {
+    fn as_any(&self) -> &dyn Any;
+    fn as_any_mut(&mut self) -> &mut dyn Any;
+}
+impl<T: Any> AsAny for T {
+    fn as_any(&self) -> &dyn Any {
+        self
+    }
 
-    /// get output text(after accept), could be override
-    fn output(&self) -> Cow<str> {
-        self.get_text()
+    fn as_any_mut(&mut self) -> &mut dyn Any {
+        self
     }
+}
 
+/// A `SkimItem` defines what's been processed(fetched, matched, previewed and returned) by skim
+///
+/// # Downcast Example
+/// Normally skim will return the item back, but in `Arc<dyn SkimItem>`. You might want a reference
```

Comment:
@alok I tried quite some time and find it difficult to accept `SkimItem` with associated type (e.g. `Receiver<T: SkimItem>`. So I turned to the `downcast` mechanism to get the concrete type information back. Though a bit complicated, does it meet your need to get different output from `SkimItem`?